packages/backend/src/webmonetization/service.ts

```diff
@@ -0,0 +1,98 @@
+import { BaseService } from '../shared/baseService'
+import { InvoiceService } from '../invoice/service'
+import { Invoice } from '../invoice/model'
+import { WebMonetization } from './model'
+import { ok } from 'assert'
+import { DateTime } from 'luxon'
+import { AccountService } from '../account/service'
+
+export interface WebMonetizationService {
+  getCurrentInvoice(accountId: string): Promise<Invoice>
+}
+
+interface ServiceDependencies extends BaseService {
+  invoiceService: InvoiceService
+  accountService: AccountService
+}
+
+export async function createWebMonetizationService({
+  logger,
+  knex,
+  invoiceService,
+  accountService
+}: ServiceDependencies): Promise<WebMonetizationService> {
+  const log = logger.child({
+    service: 'WebMonetizationService'
+  })
+  const deps: ServiceDependencies = {
+    logger: log,
+    knex,
+    invoiceService,
+    accountService
+  }
+  return {
+    getCurrentInvoice: (id) => getCurrentInvoice(deps, id)
+  }
+}
+
+async function getCurrentInvoice(
+  deps: ServiceDependencies,
+  accountId: string
+): Promise<Invoice> {
+  const account = await deps.accountService.get(accountId)
+  if (!account) {
+    throw new Error('account not found')
+  }
+
+  const wm = await WebMonetization.query(deps.knex)
+    .findById(account.id)
+    .then((wm) => {
+      if (!wm) {
+        return WebMonetization.query(deps.knex).insertAndFetch({
+          accountId: account.id
+        })
+      }
+      return wm
+    })
```

Comment:
This appears to also work
```suggestion
  const wm = await WebMonetization.query(deps.knex)
    .insertAndFetch({
      accountId: account.id
    })
    .onConflict('accountId')
    .ignore()
```